lms/djangoapps/completion/api/v1/views.py

```diff
@@ -0,0 +1,133 @@
+""" API v1 views. """
+import re
+import logging
+
+from django.http import Http404
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import permissions
+
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from opaque_keys import InvalidKeyError
+
+from lms.djangoapps.completion.models import BlockCompletion
+from openedx.core.lib.api.permissions import IsStaffOrOwner
+from student.models import CourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class CompletionBatchView(APIView):
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrOwner,)
+
+    def _validate_and_parse(self, batch_object):
+        """
+        Performs validation on the batch object to make sure it is in the proper format.
+
+        Parameters:
+            * batch_object: The object provided by the caller.
+
+        Return Value:
+            * tuple: (User, CourseKey, { block_key_1: "block_value" })
+
+        Raises:
+
+            KeyError:
+                If a key is not found in the object.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+
+            ValueError:
+                If the wrong type is passed for one of the parameters.
+        """
+        if 'username' not in batch_object:
+            raise Http404("Key 'username' not found.")
```

Comment:
One more case to check -- could you also return a 400 if the `ENABLE_COMPLETION_TRACKING` waffle switch is not enabled?
That way, people hitting the API get an error response instead of some opaque 500 error that they have to look through server logs to find.